model/src/main/kotlin/ProjectAnalyzerResult.kt

```diff
@@ -51,9 +51,14 @@ data class ProjectAnalyzerResult(
         val packageIds = packages.map { it.pkg.id }
         val referencedIds = project.collectDependencies(includeErroneous = false).map { it.id }
 
+        // Exclude project dependencies in multi-projects from the check as these appear as references in the dependency
+        // tree but not in the list of packages used.
+        val referencedIdsWithoutProjectDependencies = referencedIds.filter { it.type != "Gradle" }
```

Comment:
This class should not know about "Gradle", also "Gradle" could be used as provider for non-project dependencies in case we implement support for composite builds or source dependencies later on. I think we should either add a field to `PackageReference` to mark project dependencies or filter by project identifiers here somehow.

Reply:
> This class should not know about "Gradle"

True. That was a hack in lack of a better solution.

> I think we should either add a field to `PackageReference`

I like that idea, that would probably also make the Maven multi-module case easier to solve. Something simple like an `isProjectDependency` boolean?

Reply:
Or could we "misuse" linkage for that and introduce e.g. `PackageLinkage.PROJECT`?

Reply:
I like the idea to use linkage, but could this ever cause any conflicts? Is it possible that e.g. package managers for native code support dynamically or statically linked project dependencies?

Reply:
> Is it possible that e.g. package managers for native code support dynamically or statically linked project dependencies?

Yes, I believe that's possible. If we still stick to the `PackageLinkage` approach, possible ways to work around that include introducing `PackageLinkage.PROJECT_DYNAMIC` and `PackageLinkage.PROJECT_STATIC`, or making the linkage field a `EnumSet` so it could contain both `PackageLinkage.PROJECT` and one of `PackageLinkage.DYNAMIC` or `PackageLinkage.STATIC`.

Reply:
Alright, let's solve this issue if it ever occurs. I prefer using the linkage property over introducing a new property.